packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -108,7 +120,7 @@ export function useGridDimensions(
     } else {
       viewportOuterSize = {
         width: rootDimensionsRef.current.width,
-        height: rootDimensionsRef.current.height - totalHeaderHeight,
+        height: rootDimensionsRef.current.height - getTotalHeaderHeight(apiRef, props.headerHeight),
```

Comment:
Should we move it outside of the callback?
Otherwise, if the density changes, the callback won't be recreated since none of its dependencies has changed.

This makes me think that it would be better to explicitly pass the `density` and `maxDepth` to the `getTotalHeaderHeight` function instead of `apiRef`.
What do you think?

Reply:
The thing is that the `maxDepth` is optional. This optimization is mainly internal. `getTotalHeaderHeight` is a utility function.

Reply:
What about this one? I think that's my main concern here:

> Otherwise, if the density changes, the callback won't be recreated since none of its dependencies has changed.

Reply:
You are right I missed this one - updated.

Reply:
It should work just fine when it's outside of the callback 👍
One thing I still don't like about it is that it can be overlooked fairly easily and the value won't update.
What do you think?